### ISSUE TITLE
[REVIEW]: ggparliament: A ggplot2 extension for parliament plots in R

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1313]
**Submitting author:** @zmeers (<a href="http://orcid.org/0000-0001-8045-6531">Zoe Meers</a>)
**Repository:** <a href="https://github.com/RobWHickman/ggparliament/" target ="_blank">https://github.com/RobWHickman/ggparliament/</a>
**Version:** v.2.1.0
**Editor:** @cMadan
**Reviewer:** @andrewheiss
**Archive:** <a href="https://doi.org/10.5281/zenodo.2650709" target="_blank">10.5281/zenodo.2650709</a>

## Status

[![status](http://joss.theoj.org/papers/e90b0874dfb70891d3c2b2a781993ba9/status.svg)](http://joss.theoj.org/papers/e90b0874dfb70891d3c2b2a781993ba9)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/e90b0874dfb70891d3c2b2a781993ba9"><img src="http://joss.theoj.org/papers/e90b0874dfb70891d3c2b2a781993ba9/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/e90b0874dfb70891d3c2b2a781993ba9/status.svg)](http://joss.theoj.org/papers/e90b0874dfb70891d3c2b2a781993ba9)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/RobWHickman/ggparliament/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@andrewheiss, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @cMadan know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/RobWHickman/ggparliament/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v.2.1.0
- [x] **Authorship:** Has the submitting author (@zmeers) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @andrewheiss it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01313/joss.01313/10.21105.joss.01313.pdf)

**Comment by cMadan** [comment]
@andrewheiss, let me know if you have any questions in getting started with the review!

**Comment by cMadan** [comment]
@andrewheiss, do you know when you'll be able to begin the review? Thanks!

**Comment by andrewheiss** [comment]
Sorry for the delay! Here's my review:

---

**ggparliament** is a fantastic new package that extends **ggplot2** with custom geometries for political data. It includes a dataset of a handful of parliamentary elections and provides an easy-to-use function for creating your own legislative data. The `geom_*()` and `theme_ggparliament()` functions it provides are well documented and easy to follow, and the package's documentation in general is well done. I only came across a few issues, noted below.


## Documentation

The README runs without any problems and is easy to follow.

It might be helpful to include documentation (perhaps in the README) about what is included in the sample data. This could be as simple as adding a chunk like this to the README: 

```r
election_data %>% 
  distinct(year, country)

# or

election_data %>% 
  distinct(year, country, house)
```

The vignettes are fantastically written, are easy to follow, and they all run correctly. They should maybe be included in a list in the README so that people who come across the package through GitHub know that they exist. (Though GitHub doesn't render R Markdown nicely, unfortunately; but if the documentation is ever published online through **pkgdown** or something similar, users would be able to get to the vignettes more easily)

In the `facet_parliament_5.Rmd` vignette, there's a warning caused by line 97 because of a (vestigial?) `type = "horseshoe"` in `ggplot(aes(...))`. The `parliament_data()` function seems to take care of chart types now, so the type aesthetic doesn't do anything in `ggplot()` anymore. Removing it from `ggplot()` fixes the warning.

## Testing

There are two tests related to the structure of data created with `parliament_data()` and they pass. Adding other tests would be difficult, since they deal with visual output. **ggplot2** includes [detailed tests of its figures](https://github.com/tidyverse/ggplot2/tree/master/tests/testthat) by delving into the resulting grid objects, but that's probably not necessary here.

## Contributions

It might be helpful to have some community guidelines in the package too, such as a CONTRIBUTING.md file (perhaps modeled after something like [this](https://github.com/ropensci/nomisr/blob/master/CONTRIBUTING.md) or [this](https://github.com/tidyverse/ggplot2/blob/master/CONTRIBUTING.md)) and a CONDUCT.md file (like [this](https://github.com/ropensci/qualtRics/blob/master/CONDUCT.md) or [this](https://github.com/tidyverse/readr/blob/master/CONDUCT.md))

Creating an RStudio project `.Rproj` file could be helpful for encouraging package contributions, since it would enable RStudio's package building interface.

## DOIs

`paper/paper.bib` file should include DOIs where possible—right now, none of the entries have a DOI

---- 

That's all I have—I'm excited for this package!

**Comment by cMadan** [comment]
@andrewheiss, thank you for the well-written and comprehensive review, I really appreciate it!

@zmeers, let me know when you've had a chance to incorporate these suggestions and the review will continue from there :).

**Comment by zmeers** [comment]
@andrewheiss, thank you!

@cMadan, I have updated the README file, removed the error from one of the vignettes, and have added a CONTRIBUTING.md file. I have also included DOIs where possible in the bibliography. Thanks.

**Comment by cMadan** [comment]
@zmeers, everything looks good to me! Can you create a Github release corresponding to the current repo and tell me the new version number that now incorporates the recent changes? You also need to deposit the code somewhere archival (e.g., Zenodo or figshare) and tell me the DOI? Thanks!

**Comment by zmeers** [comment]
@cMaden, Sure, it's all done. The updated version of the R package is v.2.1.0 in the dev branch. The DOI for Zenodo is 10.5281/zenodo.2650709. Thanks!

**Comment by cMadan** [comment]
@whedon set v.2.1.0 as version

**Comment by whedon** [comment]
OK. v.2.1.0 is the version.

**Comment by cMadan** [comment]
@whedon set 10.5281/zenodo.2650709 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2650709" target="_blank">10.5281/zenodo.2650709</a> is the archive.

**Comment by cMadan** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/643

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/643, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by cMadan** [comment]
@openjournals/joss-eics, I think we're all set to accept here!

**Comment by danielskatz** [comment]
@zmeers - please add DOIs for references that have them, such as the journal papers.  See the [example files](https://joss.readthedocs.io/en/latest/submitting.html#example-paper-and-bibliography) if needed

**Comment by danielskatz** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by danielskatz** [comment]
👋 @arfon - any idea what's going on?  Either check references isn't completing, or it's very slow. Also note that it didn't complete when accept was run.

**Comment by arfon** [comment]
> 👋 @arfon - any idea what's going on? Either check references isn't completing, or it's very slow. Also note that it didn't complete when accept was run.

I'm guessing that this line might be causing failures: https://github.com/RobWHickman/ggparliament/blob/master/paper/paper.bib#L46 @zmeers - any chance you could take a look at cleaning up this entry?

**Comment by zmeers** [comment]
Hey @arfon, I believe all of the issues have been fixed in the dev branch. I've added DOIs for all articles or entries that have DOI as well. Unfortunately, a few entries do not have DOIs. You will need to compile the paper from the dev branch though, I hope that's not a problem!

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by danielskatz** [comment]
@zmeers - is there a reason you can't put these changes in the master branch?

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/644

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/644, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
☝️@whedon needs these changes to be in `master` before we can accept.

**Comment by zmeers** [comment]
@arfon, @danielskatz, it's now in the master branch! Sorry for any confusion.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/646

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/646, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
Thanks to @andrewheiss for reviewing and @cMadan for editing!

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/647
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01313](https://doi.org/10.21105/joss.01313)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01313/status.svg)](https://doi.org/10.21105/joss.01313)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01313">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01313/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01313/status.svg
   :target: https://doi.org/10.21105/joss.01313
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01313/status.svg)](https://doi.org/10.21105/joss.01313)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.18637/jss.v042.i08 is OK
- 10.18637/jss.v045.i07 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```
